### ISSUE TITLE
pylint does not detect missing attributes of builtin types

### ISSUE THREAD

**Issue by nirs** [PyCQA/pylint#1736]
```
$ cat test.py 
import io

io.BytesIO().pos

class Subclass(io.BytesIO):
    pass

Subclass().pos

class User(object):
    pass

User().pos

$ .tox/pylint/bin/pylint -E test.py 
************* Module test
E: 13, 0: Instance of 'User' has no 'pos' member (no-member)
```
Expected to get no-member error also for io.BytesIO.

**Comment by PCManticore** [comment]
Not sure what is happening, but I'm getting the correct messages emitted:

```
************* Module a
E:  3, 0: Instance of 'BytesIO' has no 'pos' member (no-member)
E:  8, 0: Instance of 'Subclass' has no 'pos' member (no-member)
E: 13, 0: Instance of 'User' has no 'pos' member (no-member)
```

**Comment by nirs** [comment]
@PCManticore, which version works?

**Comment by PCManticore** [comment]
The master version (pylint 1.8 and astroid 1.6). Can you try with it and let me know if it works for you as well? Also on python 3.

**Comment by nirs** [comment]
@PCManticore sure, we are still using 1.6.4, seems that we need to upgrade :+1:

**Comment by nirs** [comment]
Checked with latest release, broken:

$ pylint -E test.py 
************* Module test
E: 13, 0: Instance of 'User' has no 'pos' member (no-member)
(quick) [nsoffer@lean vdsm (ovirt-4.1)]$ pylint --version
pylint 1.7.5, 
astroid 1.5.3
Python 2.7.13 (default, Dec  1 2017, 09:21:53) 
[GCC 6.4.1 20170727 (Red Hat 6.4.1-1)]

Do we have 1.8 build, or should I use master?

**Comment by nirs** [comment]
Tested with pylint master (04928d6c2ac1b5bde30f2db7101dca67a8b2ffe9):
```
(py27) [nsoffer@lean pylint (master)]$ pylint --version
Using config file /home/nsoffer/src/pylint/pylintrc
pylint 1.8.0, 
astroid 1.5.3
Python 2.7.13 (default, Dec  1 2017, 09:21:53) 
[GCC 6.4.1 20170727 (Red Hat 6.4.1-1)]
(py27) [nsoffer@lean pylint (master)]$ pylint -E te
telnetlib  tempfile   termios    test       test.py    textwrap   
(py27) [nsoffer@lean pylint (master)]$ pylint -E tes
test     test.py  
(py27) [nsoffer@lean pylint (master)]$ pylint -E test.py 
Using config file /home/nsoffer/src/pylint/pylintrc
************* Module test
E: 13, 0: Instance of 'User' has no 'pos' member (no-member)
```
```
$ pylint --version
Using config file /home/nsoffer/src/pylint/pylintrc
pylint 1.8.0, 
astroid 1.5.3
Python 3.5.4 (default, Oct  9 2017, 12:07:29) 
[GCC 6.4.1 20170727 (Red Hat 6.4.1-1)]
(py35) [nsoffer@lean pylint (master)]$ pylint -E te
telnetlib  tempfile   termios    test       test.py    textwrap   
(py35) [nsoffer@lean pylint (master)]$ pylint -E tes
test     test.py  
(py35) [nsoffer@lean pylint (master)]$ pylint -E test.py 
Using config file /home/nsoffer/src/pylint/pylintrc
************* Module test
E:  3, 0: Instance of 'BytesIO' has no 'pos' member (no-member)
E:  8, 0: Instance of 'Subclass' has no 'pos' member (no-member)
E: 13, 0: Instance of 'User' has no 'pos' member (no-member)
```

So this is broken in master and python 2.7, and works in python 3.5.

I think we should reopen this issue.
